### DISCUSSION TITLE
ArgumentNullException with collection

### DISCUSSION THREAD

**Discussion by oriolaakeem** [LuckyPennySoftware/AutoMapper#3855] [NA]
ArgumentNullException: Value cannot be null. (Parameter 'property')
System.Linq.Expressions.Expression.Property(Expression expression, PropertyInfo property)
AutoMapper.Execution.ExpressionBuilder.ForEach(ParameterExpression loopVar, Expression collection, Expression loopContent)
AutoMapper.Internal.Mappers.CollectionMapper.<MapExpression>g__MapCollectionCore|2_1(Expression destExpression, ref <>c__DisplayClass2_0 )
AutoMapper.Internal.Mappers.CollectionMapper.MapExpression(IGlobalConfiguration configurationProvider, ProfileMap profileMap, MemberMap memberMap, Expression sourceExpression, Expression destExpression)
AutoMapper.Execution.ExpressionBuilder.MapExpression(IGlobalConfiguration configurationProvider, ProfileMap profileMap, TypePair typePair, Expression sourceParameter, MemberMap propertyMap, Expression destinationParameter)
AutoMapper.Execution.TypeMapPlanBuilder.MapMember(MemberMap memberMap, Expression destinationMemberValue, ParameterExpression resolvedValue)
AutoMapper.Execution.TypeMapPlanBuilder.CreatePropertyMapFunc(MemberMap memberMap, Expression destination, MemberInfo destinationMember)
AutoMapper.Execution.TypeMapPlanBuilder.CreateAssignmentFunc(Expression createDestination)
AutoMapper.Execution.TypeMapPlanBuilder.CreateMapperLambda(HashSet<TypeMap> typeMapsPath)
AutoMapper.TypeMap.Seal(IGlobalConfiguration configurationProvider, HashSet<TypeMap> typeMapsPath)
AutoMapper.MapperConfiguration.Seal()
AutoMapper.MapperConfiguration..ctor(MapperConfigurationExpression configurationExpression)
Microsoft.Extensions.DependencyInjection.ServiceCollectionExtensions+<>c.<AddAutoMapperClasses>b__12_2(IServiceProvider sp)
Autofac.Core.Activators.Delegate.DelegateActivator.ActivateInstance(IComponentContext context, IEnumerable<Parameter> parameters)
Autofac.Core.Activators.Delegate.DelegateActivator.<ConfigurePipeline>b__2_0(ResolveRequestContext ctxt, Action<ResolveRequestContext> next)
Autofac.Core.Resolving.Middleware.DisposalTrackingMiddleware.Execute(ResolveRequestContext context, Action<ResolveRequestContext> next)
Autofac.Core.Resolving.Middleware.ActivatorErrorHandlingMiddleware.Execute(ResolveRequestContext context, Action<ResolveRequestContext> next)
### Source/destination types

```csharp
// Put your source/destination types here
```

            CreateMap<NextOfKinType, NextOfKinTypeViewModel>().ReverseMap();

### Mapping configuration

```csharp
// Mapper.Initialize or just the CreateMap snippet 
```
            SourceMemberNamingConvention = new LowerUnderscoreNamingConvention();
            DestinationMemberNamingConvention = new PascalCaseNamingConvention();

### Steps to reproduce

```csharp
// Your calls to Mapper.Map or ProjectTo here, with source/destination objects constructed
```

**Comment by lbargaoanu** [comment]
A repro would help. Make a [gist](https://gist.github.com/lbargaoanu/9c7233441c3a3413cc2b9b9ebb5964a9) that we can execute and see fail.
Try the [MyGet](https://docs.automapper.org/en/latest/The-MyGet-build.html) build.

**Reply by oriolaakeem** [reply]
I only added  **ShouldMapField = fi => false;**  to the configuration and everything works fine now with 11.0.0 upgrade

**Reply by oriolaakeem** [reply]
it will be nice as default as others with old code upgrading will have to do this whenever they upgrade

**Reply by lbargaoanu** [reply]
Maybe :) But without a repro we can't tell. Some people really want to map fields, so that's just a workaround.
